package/zmq_adapter/src/zmq_adapter_tcp_connect.c

```diff
@@ -14,28 +14,55 @@
 
 #define CONNECT_RETRY_TIME_s 10
 
-static int addr_parse(const char *addr, struct sockaddr_in *s_addr)
+typedef enum
 {
-  char ip[16];
+  SUCCESS,
+  PARSE_FAILURE,
+  RESOLVE_FAILURE,
+  OTHER_ERROR
+
+} process_addr_t;
+
+static process_addr_t process_addr(const char *addr, struct sockaddr *s_addr, socklen_t *s_addr_len)
+{
+  char hostname[256] = {0};
   int port;
-  if (sscanf(addr, "%15[^:]:%d", ip, &port) != 2) {
+  if (sscanf(addr, "%255[^:]:%d", hostname, &port) != 2) {
     syslog(LOG_ERR, "error parsing address");
-    return 1;
+    return PARSE_FAILURE;
   }
 
-  memset(s_addr, 0, sizeof(*s_addr));
-  s_addr->sin_family = AF_INET;
-  s_addr->sin_port = htons(port);
+  debug_printf("connecting to %s\n", hostname);
 
-  if (inet_aton(ip, &s_addr->sin_addr) == 0) {
-    syslog(LOG_ERR, "invalid address");
-    return 1;
+  struct addrinfo *resolutions = NULL;
+  if (getaddrinfo(hostname, NULL, NULL, &resolutions) != 0) {
+    syslog(LOG_ERR, "address resolution failed");
+    return RESOLVE_FAILURE;
   }
 
-  return 0;
+  if (resolutions == NULL) {
+    syslog(LOG_ERR, "no addresses returned by name resolution");
+    return RESOLVE_FAILURE;
+  }
+
+  memcpy(s_addr, resolutions->ai_addr, resolutions->ai_addrlen);
+  freeaddrinfo(resolutions);
+
+  *s_addr_len = resolutions->ai_addrlen;
+
+  if (resolutions->ai_family == AF_INET) {
+    ((struct sockaddr_in*)s_addr)->sin_port = htons(port);
+  } else if (resolutions->ai_family == AF_INET6) {
+    ((struct sockaddr_in6*)s_addr)->sin6_port = htons(port);
+  } else {
+    syslog(LOG_ERR, "unknown address family returned from name resolution");
+    return OTHER_ERROR;
+  }
+
+  return SUCCESS;
 }
 
-static int socket_create(const struct sockaddr_in *addr)
+static int socket_create(const struct sockaddr *addr, socklen_t addr_len)
```

Comment:
Pass `addr->sa_family` to `socket` instead of the hard-coded `AF_INET`.

Reply:
Done, thanks.